examples/daal/cpp_sycl/source/logistic_regression/logistic_regression_online.hpp

```diff
@@ -0,0 +1,219 @@
+        }
+    }
+
+    // Performs the single iteration of training on the data provided by setInput() method
+    void compute()
+    {
+        namespace solver = daal::algorithms::optimization_solver::iterative_solver;
+
+        _solver->input.set(solver::inputArgument, _workPoint);
+        _solver->compute();
+
+        auto partialResult = _solver->getResult();
+        _workPoint = partialResult->get(solver::minimum);
+    }
+
+    // Creates final beta parameter values and resets internal algorithm state to default values
+    void finalizeCompute()
+    {
+        daal_dm::BlockDescriptor<Fptype> bBetas;
+        _workPoint->getBlockOfRows(0, _nBetas*_nBetaCols, daal_dm::readOnly, bBetas);
+        if (_isIntercept)
+        {
+            _finalBetas = daal_dm::SyclHomogenNumericTable<Fptype>::create(bBetas.getBuffer(), _nBetaCols, _nBetas);
+        }
+        else
+        {
+            _finalBetas = daal_dm::SyclHomogenNumericTable<Fptype>::create(_nFeatures, _nBetas, daal_dm::NumericTableIface::doAllocate);
+            daal_dm::BlockDescriptor<Fptype> bFinalBetas;
+            _finalBetas->getBlockOfRows(0, _nBetas, daal_dm::writeOnly, bFinalBetas);
+
+            auto src = bBetas.getBlockPtr();
+            auto dst = bFinalBetas.getBlockPtr();
+
+            for (size_t i = 0; i < _nBetas; i++)
+            {
+                for (size_t j = 0; j < _nFeatures; j++)
+                {
+                    dst[i*_nFeatures + j] = src[i*_nBetaCols + j + 1];
+                }
+            }
+
+            _finalBetas->releaseBlockOfRows(bFinalBetas);
+        }
+        _workPoint->releaseBlockOfRows(bBetas);
+
+        setDefaultState(_nClasses, _nBetas, _nBetaCols);
+    }
+
+    // Provides beta coefficients from the last iteration
+    auto getPartialModel() const
+    {
+        return _workPoint;
+    }
+
+    // Provides final model
+    auto getModel() const
+    {
+        daal::algorithms::logistic_regression::ModelBuilder<Fptype> mBuilder (_nFeatures, _nClasses);
+        mBuilder.setBeta(_finalBetas);
+        return mBuilder.getModel();
+    }
+
+private:
+    void setDefaultState(size_t nClasses, size_t nBetas, size_t nBetaCols)
+    {
+        _learningRateTable = daal_dm::SyclHomogenNumericTable<Fptype>::create(1, 1, daal_dm::NumericTable::doAllocate, 1.0);
+        _workPoint  = daal_dm::SyclHomogenNumericTable<Fptype>::create(1, nBetas*nBetaCols, daal_dm::NumericTable::doAllocate, 0.0);
+
+        if (nClasses != 2)
+        {
+            daal_dm::BlockDescriptor<Fptype> bd;
+            _workPoint->getBlockOfRows(0, nBetas*nBetaCols, daal_dm::writeOnly, bd);
+            auto ptr = bd.getBlockPtr();
+
+            for (size_t i = 0; i < bd.getNumberOfRows(); i += nBetaCols)
+            {
+                ptr[i] = Fptype(1e-3);
+            }
+
+            _workPoint->releaseBlockOfRows(bd);
+        }
+    }
+
+private:
+    SGDMinibatchSolverPtr _solver;
+
+    LogLossPtr _logLoss;
```

Comment:
LogLossPtr  and CrossEntropyLossPtr  have a base class objective_function::Batch why not store one instance of this pointer?

Reply:
Because as you see [here](https://github.com/oneapi-src/oneDAL/blob/e8e3085b6c69c595e1f29f13d852fe06e100bce7/examples/daal/cpp_sycl/source/logistic_regression/logistic_regression_online.hpp#L113) I must set input to the loss function. Base class do not allow to set it - `data` and `dependentVariables` are different enum values for logLoss and CrossEntropy. 
Moreover, in setParameter() I  set loss function parameters after extracting it from the solver. It is also cannot be done via base class. Rather than having the same `if`s with casts from base class to the original one I would like to make the code simpler.

Reply:
It's a pity, it would be nice.